hw01

Jasurr opened the pull request.

Jasurr committed:
test uchun fayil qoshildi
/

Jasurr committed:
for testing git

Jasurr committed:
hw01-ready for testing

Jasurr committed:
Fix: Replaced throwing RuntimeException with my custom QuestionReadException

Jasurr committed:
fix: fixed throwing errors QuestionReadException

Jasurr committed:
Merge remote-tracking branch 'origin/hw01' into hw01

stvort commented on hw01/src/main/java/ru/otus/hw/service/TestRunnerServiceImpl.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public void run() {
+        testService.executeTest();
```
Где-то все же придется обработать исключения, что мы выкидываем в дао

stvort commented on hw01/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,31 @@
+        ioService.printFormattedLine("Please answer the questions below%n");
+        // Получить вопросы из дао и вывести их с вариантами ответов
+        var questions = questionDao.findAll();
+        AtomicInteger index = new AtomicInteger(1);
+        questions.forEach(question -> {
+            ioService.printLine(index.getAndIncrement() + " - " + question.text());
+            AtomicInteger answerIndex = new AtomicInteger(0);
+            question.answers()
+                    .forEach(answer -> {
+                        char letter = (char) ('A' + answerIndex.getAndIncrement()); // A, B, C...
+                        ioService.printLine("\t" + letter + ") " + answer.text());
+                    });
+        });
```
Это лучше вытащить в метод

stvort commented on hw01/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,31 @@
+        ioService.printFormattedLine("Please answer the questions below%n");
+        // Получить вопросы из дао и вывести их с вариантами ответов
+        var questions = questionDao.findAll();
+        AtomicInteger index = new AtomicInteger(1);
+        questions.forEach(question -> {
```
Для таких вещей есть `for` со счетчиком. Не нужно любую задачу натягивать на `forEach`

stvort commented on hw01/src/main/resources/questions.csv:
```diff
@@ -0,0 +1,4 @@
+# Добавить сюда своих вопросов. Эту строку надо пропустить при настройке CsvToBean (withSkipLines)
```
Своих вопросов не добавлено

Jasurr committed:
- Added error handling to manage dao exceptions.
- Created a method to print questions and answers for better debugging and output visibility.
- Updated the questions.csv file by adding 3 new questions.

Jasurr committed:
Removed one question from the questions.csv file.

stvort commented on hw01/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,45 @@
+            var questions = questionDao.findAll();
+            printQuestions(questions);
+        } catch (QuestionReadException e) {
+            ioService.printLine(e.getMessage());
```
Выводить сообщения из исключений пользователю не стоит. Мы это обсуждали на лекции по архитектуре. Лучше тут вывести понятное сообщение основываясь на типе исключения

stvort commented on hw01/pom.xml:
```diff
@@ -0,0 +1,142 @@
+        <checkstyle-plugin.version>3.3.1</checkstyle-plugin.version>
+        <checkstyle.version>10.15.0</checkstyle.version>
+        <checkstyle.config.url>
+            https://raw.githubusercontent.com/OtusTeam/Spring/master/checkstyle.xml
```
Проверка не проходит
```
[INFO] There are 2 errors reported by Checkstyle 10.15.0 with https://raw.githubusercontent.com/OtusTeam/Spring/master/checkstyle.xml ruleset.
[ERROR] src\main\java\ru\otus\hw\service\TestServiceImpl.java:[33,34] (whitespace) WhitespaceAround: WhitespaceAround: '+' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
[ERROR] src\main\java\ru\otus\hw\service\TestServiceImpl.java:[33,34] (whitespace) WhitespaceAround: WhitespaceAround: '+' is not preceded with whitespace.
```

Jasurr committed:
fix: resolve Checkstyle issues and improve error message printing
